Jenkinsfile

```diff
@@ -0,0 +1,265 @@
+def private_key = 'dev-key'
+def vault_pass = 'vault-pass'
+def aws_creds = 'cbj-deploy'
+
+pipeline {
+  agent {
+    node {
+      label ''
+      customWorkspace 'blue-button-deploy'
+    }
+  }
+
+  parameters {
+    string(
+      defaultValue: "",
+      description: 'The branch of the application repo to deploy. Required.',
```

Comment:
does this allow us to specify a commit hash or tag?

Reply:
I'm not sure, since I haven't looked closely at how the repository is checked out.

Reply:
looks like it might work. According to ansible the git module takes a `version` argument that it looks like we're using.

> What version of the repository to check out.  This can be the the literal string HEAD, a branch name, a tag name. It can also be a SHA-1 hash, in which case refspec needs to be specified if the given revision is not already available.

If that's true then we have a way to do a really basic rollback, which is good